modules/StreamingAPI/src/main/java/org/gephi/streaming/api/GraphUpdaterEventHandler.java

```diff
@@ -0,0 +1,563 @@
+                for(Map.Entry<String, Object> entry: attributes.entrySet()) {
+                    this.addEdgeAttribute(edge, entry.getKey(), entry.getValue());
+                }
+            }
+            
+            graph.writeLock();
+            graph.addEdge(edge);
+            
+            // graph.setId(edge, edgeId);
+            graph.writeUnlock();
+        }
+    }
+    
+    private void edgeChanged(String edgeId, Map<String, Object> attributes) {
+        Edge edge = graph.getEdge(edgeId);
+        if (edge!=null) {
+            
+            graph.writeLock();
+            
+            if (attributes!=null && attributes.size() > 0) {
+                for(Map.Entry<String, Object> entry: attributes.entrySet()) {
+                    this.addEdgeAttribute(edge, entry.getKey(), entry.getValue());
+                }
+            }
+            
+            graph.writeUnlock();
+        } else {
+            log("Edge changed event ignored for edge "+edgeId+": Edge not found");
+        }
+    }
+
+    private void edgeRemoved(String edgeId) {
+        Edge edge = graph.getEdge(edgeId);
+        if (edge!=null) {
+            graph.writeLock();
+            graph.removeEdge(edge);
+            graph.writeUnlock();
+        } else {
+            log("Edge removed event ignored for edge "+edgeId+": Edge not found");
+        }
+    }
+
+    private void nodeAdded(String nodeId, Map<String, Object> attributes) {
+        Node node = graph.getNode(nodeId);
+        if (node==null) {
+            node = factory.newNode(nodeId);
+            
+            if (attributes!=null && attributes.size() > 0) {
+                for(Map.Entry<String, Object> entry: attributes.entrySet()) {
+                    this.addNodeAttribute(node, entry.getKey(), entry.getValue());
+                }
+            }
+            
+            graph.writeLock();
+            // graph.setId(node, nodeId);
+            graph.addNode(node);
+            graph.writeUnlock();
+        } else {
+            log("Node added event ignored for node "+nodeId+": Node already exists");
+        }
+    }
+    
+    private void nodeChanged(String nodeId, Map<String, Object> attributes) {
+        Node node = graph.getNode(nodeId);
+        if (node!=null) {
+            
+            graph.writeLock();
+            
+            if (attributes!=null && attributes.size() > 0) {
+                for(Map.Entry<String, Object> entry: attributes.entrySet()) {
+                    this.addNodeAttribute(node, entry.getKey(), entry.getValue());
+                }
+            }
+            
+            graph.writeUnlock();
+        } else {
+            log("Node changed event ignored for node "+nodeId+": Node not found");
+        }
+    }
+
+    private void nodeRemoved(String nodeId) {
+        Node node = graph.getNode(nodeId);
+        if (node!=null) {
+            graph.writeLock();
+
+            for (Edge edge: graph.getEdges(node).toArray()) {
```

Comment:
You don't need to do that. Removing a node will take care of removing edges. I'll update the documentation to make that clear.